packages/loot-design/src/components/modals/SelectLinkedAccounts.js

```diff
@@ -1,254 +1,200 @@
+                { name: 'External account name', width: 200 },
+                { name: 'Local account name', width: 'flex' },
+                { name: 'Actions', width: 'flex' },
+              ]}
+            />
+
+            <Table
+              items={externalAccounts}
+              style={{ backgroundColor: colors.n11 }}
+              getItemKey={index => index}
+              renderItem={({ key, item }) => (
+                <View key={key}>
+                  <TableRow
+                    externalAccount={item}
+                    linkedAccount={localAccounts.find(
+                      acc => acc.account_id === item.account_id,
+                    )}
+                    chosenAccount={
+                      chosenAccounts[item.account_id] === addAccountOption.value
+                        ? {
+                            id: addAccountOption.value,
+                            name: addAccountOption.label,
+                          }
+                        : localAccounts.find(
+                            acc => chosenAccounts[item.account_id] === acc.id,
+                          )
+                    }
+                    unlinkedAccounts={unlinkedAccounts}
+                    onSetLinkedAccount={onSetLinkedAccount}
+                  />
                 </View>
               )}
-            </View>
+            />
           </View>
-
           <View
             style={{
               flexDirection: 'row',
               justifyContent: 'flex-end',
-              marginTop: 30,
+              marginTop: 10,
             }}
           >
-            <Button style={{ marginRight: 10 }} onClick={modalProps.onClose}>
-              Cancel
-            </Button>
-            <Button primary onClick={onNext} disabled={!chosenAccounts.length}>
+            <Button
+              primary
+              onClick={onNext}
+              disabled={!Object.keys(chosenAccounts).length}
+            >
               Link accounts
             </Button>
           </View>
-        </View>
+        </>
       )}
     </Modal>
   );
 }
+
+function TableRow({
+  externalAccount,
+  linkedAccount,
+  chosenAccount,
+  unlinkedAccounts,
+  onSetLinkedAccount,
+}) {
+  const [focusedField, setFocusedField] = useState(null);
+
+  const chosenAccountOption = chosenAccount && {
+    value: chosenAccount.id,
+    label: chosenAccount.name,
+  };
+
+  const availableAccountOptions = [
+    ...unlinkedAccounts.map(acct => ({
+      value: acct.id,
+      label: acct.name,
+    })),
+    chosenAccount?.id !== addAccountOption.value && chosenAccountOption,
+    addAccountOption,
+  ].filter(Boolean);
+
+  return (
+    <Row style={{ backgroundColor: 'white' }}>
+      <Field width={200}>{externalAccount.name}</Field>
+      <Field
+        width="flex"
+        truncate={focusedField !== 'account'}
+        onClick={() => !linkedAccount && setFocusedField('account')}
+      >
+        {focusedField === 'account' ? (
+          <Autocomplete
```

Comment:
In the screenshot it looks like this field has a significant misalignment with the column title — could they maybe be matched somehow?

Reply:
Yea, that's actually already fixed.. I was just lazy and didn't update the screenshots in the PR :P